src/Ocelot/DownstreamRouteFinder/Middleware/DownstreamRouteFinderMiddleware.cs

```diff
@@ -34,7 +34,7 @@ public async Task Invoke(HttpContext httpContext)
 
             var upstreamQueryString = httpContext.Request.QueryString.ToString();
 
-            var upstreamHost = httpContext.Request.Headers["Host"];
+            var upstreamHost = httpContext.Request.Headers["Host"].ToString().Split(":")[0];
```

Comment:
Better to test for colon sign `:` first to ensure that a port is specified. 👇 

```suggestion
            var hostHeader = httpContext.Request.Headers["Host"].ToString();
            var upstreamHost = hostHeader.Contains(':')
                ? hostHeader.Split(':')[0]
                : hostHeader;
```